File: teleticwebsiteback/src/main/java/com/teleticwebsiteback/teleticwebsiteback/model/Kit.java
package com.teleticwebsiteback.teleticwebsiteback.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Kit {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "ID")
    private int id;

    @OneToOne
    @JoinColumn(name = "tpe_fk")
    private Tpe tpe;

    @OneToOne
    @JoinColumn(name = "sim_fk")
    private Sim sim;

    @OneToOne
    @JoinColumn(name = "store_fk")
    private Store store;

    @OneToOne
    @JoinColumn(name = "consumable_fk")
    private Consumable conso;


}

File: src/app/model/Kit.ts
export class Kit {
    id: number;
    tpe: number;
    sim: number;
    store: number;
    conso: number;

    constructor(id, tpe, sim, store, conso) {
        this.id = id;
        this.tpe = tpe;
        this.sim = sim;
        this.store = store;
        this.conso = conso;
    }
}
